simpleclient_spring_web/src/main/java/io/prometheus/client/spring/web/PrometheusTimeMethods.java

```diff
@@ -0,0 +1,46 @@
+package io.prometheus.client.spring.web;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Enable Spring-AOP-based automated method timing for the annotated method or all methods in an annotated class.
+ * The timings will be recorded in a Histogram with the default metric name of {@code prometheus_method_timing},
+ * which can be overridden by providing a {@code value} parameter in the annotation, and optionally a {@code help}
+ * parameter and a single label of {@code signature} whose values will be {@code ClassOrInterfaceName.methodName(..)}.
+ *
+ * To properly work, {@link EnablePrometheusTiming} must be specified somewhere in your application configuration.
+ *
+ *  <pre><code>
+ * {@literal @}PrometheusTimeMethods("my_app_timer_seconds")
```

Comment:
> So were they seeing performance issues in the application itself, due to Prometheus overhead? 

That wasn't reported to me, but I presume not. Though with this level of instrumentation there's likely an inner loop somewhere where it'd cause a problem.

> like the better experience, as long as it goes with the warning that "hey, use metrics sparingly," which seems like it applies regardless of whether the metrics were created with annotations, or by adding all the requisite code themselves.

The problem is that metrics are basically free, as long as you don't use them in an industrial fashion. Class level annotations tip the scales for me where it's too easy to run yourself into problems.

If you really want per-function data for every single function, that's more the job for a profiling tool than a metrics one.